Game/Pregnancy/MenstrualCycle.gd

```diff
@@ -190,18 +218,26 @@ func getPregnancyProgress() -> float:
 	#	print("PREGNANCY: "+str(maxProgress))
 	return maxProgress
 
-func isReadyToGiveBirth():
-	return getPregnancyProgress() >= 1.0
+func isReadyToGiveBirth() -> bool:
+	var readyFetusAmount: = 0
+	for egg in impregnatedEggCells:
+		if(egg.fetusIsReadyForBirth()):
+			readyFetusAmount += 1
+			
+	if(impregnatedEggCells.size() > 0):
+		return readyFetusAmount == impregnatedEggCells.size()
```

Comment:
I usually expand it into the full if(a == b) return true else return false. Easier for me to read.
I also suggest replacing == with >=. Yeah, it won't change anything but it feels better for me
Also just a note but this means you can technically go from 'being ready to give birth' back to not being ready if one of the eggcells got impregnated late. Just wondering if that's intended

Reply:
It is intended yeah.